test/SlowTests/Issues/RavenDB-11795.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Linq;
+using FastTests;
+using Raven.Client.Documents.Indexes;
+using Xunit;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_11795 : RavenTestBase
+    {
+        [Fact]
+        public void SearchBooking_ProjectionWithDateTimeToStringAndFormat_ReturnsResult()
+        {
+            using (var store = GetDocumentStore())
+            { 
+                // Arrange  
+                store.ExecuteIndex(new BookingIndex());
+
+                using (var session = store.OpenSession())
+                {
+                    session.Advanced.WaitForIndexesAfterSaveChanges(TimeSpan.FromSeconds(2));
+
+                    session.Store(new Booking { FirstName = "Alex", LastName = "Me", Start = DateTime.Parse("2018-01-01T11:11:11") });
+                    session.Store(new Booking { FirstName = "You", LastName = "Me", Start = DateTime.Parse("2017-11-11T10:10:10") });
+                    session.SaveChanges();
+                }
+
+                WaitForIndexing(store);
+
+                // Act
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Query<BookingIndex.Result, BookingIndex>()
+                        .Where(x => x.FullName == "Alex Me")
+                        .Select(x => new {
+                            FullName = x.FullName,
+                            StartDate = x.Start.ToString("dd.MM.yyyy")
+                        });
+
+                    Assert.Equal("from index 'BookingIndex' as x where x.FullName = $p0 " +
+                                 "select { FullName : x.FullName, StartDate : toDateString(new Date(Date.parse(x.Start)), \"dd.MM.yyyy\") }"
```

Comment:
Will `toDateString()` support _any_ custom format that you can use in c#? (https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings)

Reply:
it should, it is implemented in c# in `ScriptRunner` :
https://github.com/ravendb/ravendb/blob/be4c5d0d81754bb0747a0f966302b56202673a9d/src/Raven.Server/Documents/Patch/ScriptRunner.cs#L653-L656
should I add more tests with other formats?